tracer/src/Datadog.Trace/SpanExtensions.cs

```diff
@@ -24,14 +24,9 @@ public static void SetUser(this ISpan span, UserDetails userDetails)
                 ThrowHelper.ThrowArgumentNullException(nameof(span));
             }
 
-            if (userDetails is null)
+            if (string.IsNullOrEmpty(userDetails.Id))
```

Comment:
nit: It won't let me add comments on this PR further down but the `userDetails.Id is not null` check is not required 🙂